tests/components/tesla_fleet/test_climate.py

```diff
@@ -0,0 +1,403 @@
+
+    # Call set temp without temperature
+    with pytest.raises(
+        ServiceValidationError, match="Temperature is required for this action"
+    ):
+        await hass.services.async_call(
+            CLIMATE_DOMAIN,
+            SERVICE_SET_TEMPERATURE,
+            {
+                ATTR_ENTITY_ID: [entity_id],
+                ATTR_TARGET_TEMP_HIGH: 30,
+                ATTR_TARGET_TEMP_LOW: 30,
+            },
+            blocking=True,
+        )
+    state = hass.states.get(entity_id)
+    assert state.attributes[ATTR_TEMPERATURE] == 40
+    assert state.state == HVACMode.COOL
+
+    # Call set temp with invalid temperature
+    with pytest.raises(
+        ServiceValidationError,
+        match="Cabin overheat protection does not support that temperature",
+    ):
+        # Invalid Temp
+        await hass.services.async_call(
+            CLIMATE_DOMAIN,
+            SERVICE_SET_TEMPERATURE,
+            {ATTR_ENTITY_ID: [entity_id], ATTR_TEMPERATURE: 34},
+            blocking=True,
+        )
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_climate_alt(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    mock_vehicle_data: AsyncMock,
+    normal_config_entry: MockConfigEntry,
+) -> None:
+    """Tests that the climate entity is correct."""
+
+    mock_vehicle_data.return_value = VEHICLE_DATA_ALT
+    await setup_platform(hass, normal_config_entry, [Platform.CLIMATE])
+    assert_entities(hass, normal_config_entry.entry_id, entity_registry, snapshot)
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_climate_offline(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    mock_vehicle_data: AsyncMock,
+    normal_config_entry: MockConfigEntry,
+) -> None:
+    """Tests that the climate entity is correct."""
+
+    mock_vehicle_data.side_effect = VehicleOffline
+    await setup_platform(hass, normal_config_entry, [Platform.CLIMATE])
+    assert_entities(hass, normal_config_entry.entry_id, entity_registry, snapshot)
+
+
+async def test_invalid_error(
+    hass: HomeAssistant,
+    normal_config_entry: MockConfigEntry,
+) -> None:
+    """Tests service error is handled."""
+
+    await setup_platform(hass, normal_config_entry, platforms=[Platform.CLIMATE])
+    entity_id = "climate.test_climate"
+
+    with (
+        patch(
+            "homeassistant.components.tesla_fleet.VehicleSpecific.auto_conditioning_start",
+            side_effect=InvalidCommand,
+        ) as mock_on,
+        pytest.raises(HomeAssistantError) as error,
+    ):
+        await hass.services.async_call(
+            CLIMATE_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: [entity_id]},
+            blocking=True,
+        )
+    mock_on.assert_called_once()
+    assert str(error.value) == "Command failed: The data request or command is unknown."
```

Comment:
Move into `pytest.raises` as the `match` argument here and below

Reply:
Thanks, this helped identify a bug in one of my error strings too.